scripts/dead_code_detector.py

```
@@ -0,0 +1,408 @@
+        for js_file in self.project_root.glob('frontend/src/**/*.{js,jsx,ts,tsx}'):
+            with open(js_file, 'r') as f:
+                content = f.read()
+                for match in re.finditer(pattern, content):
+                    imports.append(match.group(1))
+
+        # Find unused components (excluding common ones)
+        common_components = {'React', 'useState', 'useEffect', 'useRef', 'useCallback', 
+                           'useMemo', 'useContext', 'useReducer', 'useLayoutEffect',
+                           'useImperativeHandle', 'useDebugValue', 'useDeferredValue',
+                           'useTransition', 'useId', 'createContext', 'Fragment',
+                           'Suspense', 'lazy', 'memo', 'forwardRef', 'useRef'}
```

Ruff (0.15.21)

[warning] Sets should not contain duplicate item `'useRef'`

Remove duplicate item

(B033)